src/stream/src/executor/hash_join.rs

```diff
@@ -622,12 +616,32 @@ impl<K: HashKey, S: StateStore, const T: JoinTypePrimitive> HashJoinExecutor<K,
             (side_update.start_pos, side_match.start_pos)
         };
 
+        let update_range = update_start_pos..(update_start_pos + side_update.col_types.len());
+        let matched_range = matched_start_pos..(side_match.col_types.len() + matched_start_pos);
+
+        let (update_range, matched_range) = if T == LeftAnti || T == LeftSemi {
+            if SIDE == SideType::Left {
+                (update_range, 0..0)
+            } else {
+                (0..0, matched_range)
+            }
+        } else if T == RightAnti || T == RightSemi {
+            if SIDE == SideType::Left {
+                (0..0, matched_range)
+            } else {
+                (update_range, 0..0)
+            }
+        } else {
+            (update_range, matched_range)
+        };
```

Comment:
The code here is kind of complex. :innocent: It mainly focuses on semi/anti joins. 

For left semi/anti joins, only the range of the left side should be full. And the range of the other side should be $\varnothing$.